scrapy_zyte_api/handler.py

```diff
@@ -66,10 +70,16 @@ def download_request(self, request: Request, spider: Spider) -> Deferred:
         return super().download_request(request, spider)
 
     def _prepare_api_params(self, request: Request) -> Optional[dict]:
-        meta_params = request.meta.get("zyte_api")
+        meta_params = request.meta.get("zyte_api", self._all)
+        if meta_params is False:
+            return None
         if not meta_params and meta_params != {}:
+            warn(
+                f"Setting the zyte_api request metadata key to "
+                f"{meta_params!r} is deprecated. Use False instead.",
```

Comment:
If the `meta_params` value is set to `False`, this message would be triggered.

Since we have this in the README:
> Set the `ZYTE_API_ALL` setting to `True` to make all requests go through Zyte Data API unless the `zyte_api` key in [Request.meta](https://docs.scrapy.org/en/latest/topics/request-response.html#scrapy.http.Request.meta) is set to `False`.

then I think this warning message might be confusing.

Do we mean somethin else here?

Reply:
> If the `meta_params` value is set to `False`, this message would be triggered.

It shouldn’t due to the early return above.

Reply:
Ahhh gotcha, we're checking `None` here. 👍

Reply:
I did not limit it to `meta_params is None` because the original code was not checking that either, so it is technically possible for someone to use `"zyte_api": 0` or similar. I don’t think anyone ever did, but just in case.